src/gui/attributetable/qgsfeaturelistview.cpp

```diff
@@ -204,14 +204,20 @@ void QgsFeatureListView::setEditSelection( const QModelIndex &index, QItemSelect
 {
   bool ok = true;
   emit aboutToChangeEditSelection( ok );
-
-#ifdef QGISDEBUG
-  if ( index.model() != mModel->masterModel() )
+  
+  if ( index.isValid() && index.model() != mModel->masterModel() )
```

Comment:
Is there a reason for the additional conditional code changes compared to just adding the `index.isValid()` condition inside warning?

For what I can see it doesn't make any difference in reality but makes the code harder to read because it looks like a deliberate choice.

Reply:
Also -- isn't this a bad bug, if something results in passing a index from the wrong model? Why is this happening in the first place?

Reply:
Yes, normally. That's why there is the warning, which therefore IMO should be the _only_ thing within this branch.  
There is one valid case though which is an invalid index, where model is `nullptr`.